actionpack/lib/action_dispatch/system_testing/browser.rb

```diff
@@ -70,6 +69,15 @@ def set_headless_firefox_browser_options
             capabilities.add_argument("-headless")
           end
         end
+
+        def resolve_driver_path(namespace)
+          service = namespace::Service
+          service.driver_path ||= ::Selenium::WebDriver::DriverFinder.path(
+            options || namespace::Options.new,
+            service
+          )
+          service.driver_path.try(:call)
```

Comment:
Looking at the source code for `DriverFinder.path` [in `selenium-webdriver` 4.11](https://github.com/SeleniumHQ/selenium/blob/selenium-4.11.0/rb/lib/selenium/webdriver/common/driver_finder.rb#L23-L25) (for Ruby >= 3) and [in `selenium-webdriver` 4.9](https://github.com/SeleniumHQ/selenium/blob/selenium-4.9.0/rb/lib/selenium/webdriver/common/driver_finder.rb#L23-L25) (for Ruby < 3), I think we can simply set `driver_path` to its return value.

`DriverFinder.path` will use `Service.driver_path` if present, eliminating the need for `||=`.  It will also evaluate the `Proc` if necessary, eliminating the need for `service.driver_path.try(:call)`.

```suggestion
          namespace::Service.driver_path = ::Selenium::WebDriver::DriverFinder.path(
            options || namespace::Options.new,
            namespace::Service
          )
```

What do you think?

Reply:
Great suggestion. Thanks!